standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3343,28 +3343,36 @@ private void ensureAllTxnsValid(List<Long> txnIds, Statement stmt) throws SQLExc
             txnIds, "txn_id", false, false);
     Long txnId;
     char txnState;
+    boolean isAborted = false;
+    StringBuilder errorMsg = new StringBuilder("Write ID allocation failed for input txns: ");
     for (String query : queries) {
       LOG.debug("Going to execute query <" + query + ">");
       ResultSet rs = stmt.executeQuery(query);
-      if (rs.next()) {
+      while (rs.next()) {
         txnId = rs.getLong(1);
         txnState = rs.getString(2).charAt(0);
         if (txnState != TXN_OPEN) {
-          throw new IllegalStateException("Write ID allocation failed as input txn: " + txnId
-                  + " with state: " + txnState + " is not in open state.");
+          isAborted = true;
+          errorMsg.append("{").append(txnId).append(",").append(txnState).append("}");
         }
       }
     }
     // Check if any of the txns in the list is committed.
-    checkIfTxnsCommitted(txnIds, stmt);
+    boolean isCommitted = checkIfTxnsCommitted(txnIds, stmt, errorMsg);
+    if (isAborted || isCommitted) {
+      LOG.error(errorMsg.toString());
```

Comment:
nit: it would be very useful to include dbname.table name in msg/exception

Reply:
Fixed.